### PR TITLE
ci: changelog modifications aren't overwritten after next releases

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #

## Summary
It turns out the default behavior of conventional-changelog is to generate the changelog items since the most recent tag. Whoever wrote the script was deleting all of the next tags and readding them so that the whole changelog would be regenerated. I ended up writing some helpers to combine the existing unreleased changelog items and the new ones.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,45 +10,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### ⚠ BREAKING CHANGES
 
 - **list, list-item, list-item-group:** To know when `calcite-list-item` content is selected, listen to the event `calciteListItemSelect` instead of `click`.
-
```
Comment (driskull) [inline CHANGELOG.md:13]: @benelan why are you removing this?

Comment (driskull) [reply CHANGELOG.md:13]: This part should stay.

**Review by driskull — APPROVED** [review]
👍

File: CHANGELOG.md (hunk 1 above)
Comment (benelan) [reply CHANGELOG.md:13]: Oh okay I thought it was accidently added like Eliza's was. I'll put it back

**Pull request merged**